### ISSUE TITLE
Unexpected Attribute Error for Valid Parameters 

### ISSUE THREAD

**Issue by Irori235** [terraform-google-modules/terraform-google-kubernetes-engine#1804]
### TL;DR

While using the Terraform extension in VSCode, I'm encountering an "Unexpected attribute" error for valid value such as `region`. 

Initially, I susupected the Terraform language server might be the coused, but similar bug dosen't seem to occur with other modules like `terraform-google-modules/network/google`.

```
module "gke" {
  source                     = "terraform-google-modules/kubernetes-engine/google"
  version                    = "29.0.0"
  project_id                 = var.google_cloud_project
  name                       = var.gke_cluster_name
  region                     = var.google_cloud_region
  zones                      = [var.gke_zone]
  network                    = module.vpc.network_name
  subnetwork                 = module.vpc.subnets_names[0]
  ip_range_pods              = "gke-pods"
  ip_range_services          = "gke-services"
  http_load_balancing        = true
  horizontal_pod_autoscaling = true
  regional                   = true


  node_pools = [
    {
      name         = "default-node-pool"
      machine_type = "e2-small"
      min_count    = 2
      max_count    = 3
      spot         = true
      auto_repair  = true
      auto_upgrade = true
    }
  ]

  node_pools_labels = {
    all = {}

    default-node-pool = {
      default-node-pool = true
    }
  }

  node_pools_metadata = {
    all = {}

    default-node-pool = {
      node-pool-metadata-custom-value = "my-node-pool"
    }
  }

  node_pools_taints = {
    all = []

    default-node-pool = [
      {
        key    = "default-node-pool"
        value  = true
        effect = "PREFER_NO_SCHEDULE"
      },
    ]
  }

  node_pools_tags = {
    all = []

    default-node-pool = [
      "default-node-pool",
    ]
  }

}
```

<img width="541" alt="Screenshot_Nov19" src="https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/84654022/92715abc-9630-4177-9bb4-0f6865b49502">

### Expected behavior

No error for valid value 

### Terraform Configuration

```hcl
"terraform.codelens.referenceCount": true,
  "terraform.experimentalFeatures.validateOnSave": true,
  "terraform.languageServer.enable": true,
  "terraform.languageServer.args": [
    "serve"
  ],
```

### Terraform Version

```sh
Terraform v1.6.3
```

### Additional information

- I can read the parameter descriptions, and the autocomplete is working similarly well.
- I have tried reinstalling VSCode, terraform, terraform extensions, restarting terraform language server, but no change.
